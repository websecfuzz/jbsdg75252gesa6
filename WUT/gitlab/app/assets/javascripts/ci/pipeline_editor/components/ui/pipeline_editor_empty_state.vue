<script>
import { GlButton, GlSprintf, GlEmptyState } from '@gitlab/ui';
import { __ } from '~/locale';
import PipelineEditorFileNav from '~/ci/pipeline_editor/components/file_nav/pipeline_editor_file_nav.vue';

export default {
  components: {
    GlButton,
    GlSprintf,
    GlEmptyState,
    PipelineEditorFileNav,
  },
  i18n: {
    title: __('Optimize your workflow with CI/CD Pipelines'),
    body: __(
      'Create a new %{codeStart}.gitlab-ci.yml%{codeEnd} file at the root of the repository to get started.',
    ),
    btnText: __('Configure pipeline'),
    externalCiNote: __("This project's pipeline configuration is located outside this repository"),
    externalCiInstructions: __(
      'To edit the pipeline configuration, you must go to the project or external site that hosts the file.',
    ),
  },
  inject: {
    emptyStateIllustrationPath: {
      default: '',
    },
    usesExternalConfig: {
      default: false,
      type: Boolean,
      required: false,
    },
  },
  methods: {
    createEmptyConfigFile() {
      this.$emit('createEmptyConfigFile');
    },
  },
};
</script>
<template>
  <div>
    <pipeline-editor-file-nav v-on="$listeners" />
    <gl-empty-state
      v-if="usesExternalConfig"
      :title="$options.i18n.externalCiNote"
      :description="$options.i18n.externalCiInstructions"
      :svg-path="emptyStateIllustrationPath"
    />

    <gl-empty-state v-else :title="$options.i18n.title" :svg-path="emptyStateIllustrationPath">
      <template #description>
        <gl-sprintf :message="$options.i18n.body">
          <template #code="{ content }">
            <code>{{ content }}</code>
          </template>
        </gl-sprintf>
      </template>
      <template #actions>
        <gl-button
          variant="confirm"
          class="gl-mt-3"
          data-testid="create-new-ci-button"
          @click="createEmptyConfigFile"
        >
          {{ $options.i18n.btnText }}
        </gl-button>
      </template>
    </gl-empty-state>
  </div>
</template>
