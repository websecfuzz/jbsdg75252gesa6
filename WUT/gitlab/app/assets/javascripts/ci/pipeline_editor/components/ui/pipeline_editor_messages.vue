<script>
import { GlAlert } from '@gitlab/ui';
import { getParameterValues, removeParams } from '~/lib/utils/url_utility';
import { __, s__ } from '~/locale';
import {
  COMMIT_FAILURE,
  DEFAULT_FAILURE,
  LOAD_FAILURE_UNKNOWN,
  PIPELINE_FAILURE,
} from '../../constants';
import CodeSnippetAlert from '../code_snippet_alert/code_snippet_alert.vue';
import {
  CODE_SNIPPET_SOURCE_URL_PARAM,
  CODE_SNIPPET_SOURCES,
} from '../code_snippet_alert/constants';

export default {
  components: {
    GlAlert,
    CodeSnippetAlert,
  },

  errors: {
    [COMMIT_FAILURE]: s__('Pipelines|The GitLab CI configuration could not be updated.'),
    [DEFAULT_FAILURE]: __('Something went wrong on our end.'),
    [LOAD_FAILURE_UNKNOWN]: s__('Pipelines|The CI configuration was not loaded, please try again.'),
    [PIPELINE_FAILURE]: s__('Pipelines|There was a problem with loading the pipeline data.'),
  },
  props: {
    failureType: {
      type: String,
      required: false,
      default: null,
    },
    failureReasons: {
      type: Array,
      required: false,
      default: () => [],
    },
    showFailure: {
      type: Boolean,
      required: false,
      default: false,
    },
  },
  data() {
    return {
      codeSnippetCopiedFrom: '',
    };
  },
  computed: {
    failure() {
      const { errors } = this.$options;

      return {
        text: errors[this.failureType] ?? errors[DEFAULT_FAILURE],
        variant: 'danger',
      };
    },
  },
  created() {
    this.parseCodeSnippetSourceParam();
  },
  methods: {
    dismissCodeSnippetAlert() {
      this.codeSnippetCopiedFrom = '';
    },
    dismissFailure() {
      this.$emit('hide-failure');
    },
    parseCodeSnippetSourceParam() {
      const [codeSnippetCopiedFrom] = getParameterValues(CODE_SNIPPET_SOURCE_URL_PARAM);
      if (codeSnippetCopiedFrom && CODE_SNIPPET_SOURCES.includes(codeSnippetCopiedFrom)) {
        this.codeSnippetCopiedFrom = codeSnippetCopiedFrom;
        window.history.replaceState(
          {},
          document.title,
          removeParams([CODE_SNIPPET_SOURCE_URL_PARAM]),
        );
      }
    },
  },
};
</script>

<template>
  <div>
    <code-snippet-alert
      v-if="codeSnippetCopiedFrom"
      :source="codeSnippetCopiedFrom"
      class="gl-mb-5"
      @dismiss="dismissCodeSnippetAlert"
    />
    <gl-alert
      v-if="showFailure"
      :variant="failure.variant"
      class="gl-mb-5"
      @dismiss="dismissFailure"
    >
      {{ failure.text }}
      <ul v-if="failureReasons.length" class="gl-mb-0">
        <li v-for="reason in failureReasons" :key="reason">{{ reason }}</li>
      </ul>
    </gl-alert>
  </div>
</template>
