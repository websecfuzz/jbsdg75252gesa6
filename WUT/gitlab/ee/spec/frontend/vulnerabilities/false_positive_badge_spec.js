import { GlBadge } from '@gitlab/ui';
import { shallowMount } from '@vue/test-utils';
import FalsePositiveBadge from 'ee/vulnerabilities/components/false_positive_badge.vue';

const TITLE = 'False positive detected';
const MESSAGE =
  'The scanner determined this vulnerability to be a false positive. Verify the evaluation before changing its status. %{linkStart}Learn more about false positive detection.%{linkEnd}';

describe('False positive badge component', () => {
  let wrapper;

  const createWrapper = (provide) => {
    return shallowMount(FalsePositiveBadge, {
      provide: {
        canViewFalsePositive: true,
        ...provide,
      },
    });
  };

  it('should render the alert badge', () => {
    wrapper = createWrapper();

    const { i18n } = wrapper.vm.$options;

    expect(i18n.title).toEqual(TITLE);
    expect(i18n.message).toEqual(MESSAGE);

    expect(wrapper.element).toMatchSnapshot();
  });

  it('should not render the alert badge when canViewFalsePositive is false', () => {
    wrapper = createWrapper({ canViewFalsePositive: false });

    expect(wrapper.findComponent(GlBadge).exists()).toEqual(false);
  });
});
