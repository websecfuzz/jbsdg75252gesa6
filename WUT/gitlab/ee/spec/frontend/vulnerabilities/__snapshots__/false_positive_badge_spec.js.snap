// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`False positive badge component should render the alert badge 1`] = `
<div
  class="gl-inline-block"
>
  <gl-badge-stub
    icon="false-positive"
    iconsize="md"
    tag="span"
    target="_self"
    variant="warning"
  />
  <gl-popover-stub
    cssclasses=""
    placement="top"
    target="[Function]"
    title="False positive detected"
  >
    <gl-sprintf-stub
      message="The scanner determined this vulnerability to be a false positive. Verify the evaluation before changing its status. %{linkStart}Learn more about false positive detection.%{linkEnd}"
    />
  </gl-popover-stub>
</div>
`;
